src/common/lib/api.ts

```diff
@@ -33,64 +29,76 @@ export interface AccountInfoRes {
   name: string;
 }
 
-interface StatusRes {
+export interface StatusRes {
   configured: boolean;
   unlocked: boolean;
   currentAccountId: string;
 }
 
-interface UnlockRes {
+export interface UnlockRes {
   currentAccountId: string;
 }
 
 interface BlocklistRes {
   blocked: boolean;
 }
 
-export const getAccountInfo = () => msg.request<AccountInfoRes>("accountInfo");
-
-/**
- * stale-while-revalidate get account info
- * @param id - account id
- * @param callback - will be called first with cached (stale) data first, then with fresh data.
- */
-export const swrGetAccountInfo = async (
-  id: string,
-  callback?: (account: AccountInfo) => void
-): Promise<AccountInfo> => {
-  const accountsCache = await getAccountsCache();
-
-  return new Promise((resolve, reject) => {
-    if (accountsCache[id]) {
-      if (callback) callback(accountsCache[id]);
-      resolve(accountsCache[id]);
-    }
-
-    // Update account info with most recent data, save to cache.
-    getAccountInfo()
-      .then((response) => {
-        const { alias } = response.info;
-        const { balance: resBalance, currency } = response.balance;
-        const name = response.name;
-        const balance =
-          typeof resBalance === "number" ? resBalance : parseInt(resBalance); // TODO: handle amounts
-        const account = {
-          id,
-          name,
-          alias,
-          balance,
-          currency: currency || "BTC", // set default currency for every account
-        };
-        storeAccounts({
-          ...accountsCache,
-          [id]: account,
-        });
-        if (callback) callback(account);
-        return resolve(account);
-      })
-      .catch(reject);
-  });
+export const getAccountInfo = (): Promise<AccountInfoRes> =>
+  msg.request<AccountInfoRes>("accountInfo");
+
+export const getAccountInfoKey = (
+  id: string | null | undefined
+): string | null => (id ? `accountInfo/${id}` : null);
+
+// useSWR hook is expected to be called in every render, hence we expect parameter id null to skip fetching
+export const useAccountInfoCached = async (
+  id: string | null
+): Promise<AccountInfo | null> => {
+  const { data, error } = await useSWR<AccountInfoRes>(
```

Comment:
can you return the mutate function here too? unless you need to mutate an account that is not currently selected?

Reply:
Yes, good point. By refactoring the `useSWR` to the component level I can easily use bound mutate. This really simplifies things.